packages/react/src/query/prepareSwap.ts

```diff
@@ -0,0 +1,38 @@
+import type { Config } from "wagmi";
+import type { QueryOptions } from "@tanstack/react-query";
+import { filterQueryOptions } from "./utils.js";
+import {
+  prepareSwap,
+  type PrepareSwapParameters,
+  type PrepareSwapReturnType,
+} from "@alchemy/wagmi-core";
+
+// TODO(jh): may want to use `UnionExactPartial<PrepareSwapParameters>` here. test it first.
+export type PrepareSwapOptions = PrepareSwapParameters;
+
+function prepareSwapQueryKey(options: PrepareSwapOptions) {
+  const { connector: _connector, ...rest } = options;
+  return ["prepareSwap", filterQueryOptions(rest)] as const;
+}
+
+export type PrepareSwapQueryKey = ReturnType<typeof prepareSwapQueryKey>;
+
+export function prepareSwapQueryOptions(
+  config: Config,
+  options: PrepareSwapOptions,
+) {
+  return {
+    async queryFn({ queryKey }) {
+      const { connector } = options;
+      const { ...params } = queryKey[1];
```

Comment:
nit: I think what you have may still work but should this just be `const params = queryKey[1];`